src/incubator/Slider/Thumb.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import {StyleSheet, ViewProps} from 'react-native';
+import {GestureDetector} from 'react-native-gesture-handler';
+import {Colors} from '../../style';
+import View from '../../components/view';
+
+const THUMB_SIZE = 24;
+const THUMB_BORDER_WIDTH = 6;
+const SHADOW_RADIUS = 4;
+
+export interface Props extends ViewProps {
+  animatedStyle?: any;
+  gesture?: any;
+}
+
+const Thumb = (props: Props) => {
```

Comment:
You can extract much more logic and code into this component. This is not enough. 
There are still lots of similar duplicate code for default and range thumbs in the main file. 

Try to isolate the Thumb component and think of it as a standalone. 

From what I see this component should accept
- value
- available range to move in (For a simple Slider we should pass the whole slider range. For a range slider, this value should always change)

Once you move the relevant data to this component you'll be able to move most of the code from the main file. Style, gesture, etc..

Reply:
If I move the gesture logic to the thumb then I will have to have callbacks on every phase so the main Slider can update the track and other dependent logic, and will have to keep updating the Thumb's value. This will create code duplications other than reducing the length of the code...